### ISSUE TITLE
【要望】定数の英語名

### ISSUE THREAD

**Issue by hori-ryota** [kabucom/kabusapi#12]
各定数について言語、ライブラリ間の統一のためドキュメントに英語名を付記できないでしょうか。
現状ですと各々の開発者が翻訳した英語名もしくはローマ字にて実装されるため、互換性が低くなりがちかと思います。

日本でしかない概念の用語はローマ字でもよいと思うので、変数名に使いやすい英語名を付記いただけると自動生成にも活用しやすく助かります。

**Comment by yasuyuki-nakazawa** [comment]
ご要望に関しては対応していきます。
リファレンスへの英語名の追加は順次対応。
ただし、変数名に関しては検討させていただきます。

**Comment by hori-ryota** [comment]
要望については回答ではなく、対応もしくは対応しないの決めのタイミングでcloseしたほうがわかりやすいかもですね（もちろん運用、管理しやすいやり方を優先していただければ異論はないです）

**Comment by yasuyuki-nakazawa** [comment]
リオープンいたします。対応完了でCloseいたします。

**Comment by hori-ryota** [comment]
ありがとうございます！

**Comment by yasuyuki-nakazawa** [comment]
こちら、対応までお時間かかっておりますが、要望としては、現在のリファレンスの「説明」部分で、
現行の記載列の右側に英語記載があればよいというかたちでしょうか？

**Comment by hori-ryota** [comment]
はい、形式は特に問わないので各要素について命名に使える英語名を付記いただけますと互換性が高まるかと思います。ありがとうございます。

**Comment by yasuyuki-nakazawa** [comment]
証券関連の用語は英語化すると、非常に長くなる傾向があり、定数としてそのまま利用するというのには適切ではないとおもわれますが、単語を元に、定数定義する参考とするという目的にはお答えできるかと思います
いかがでしょうか。

**Comment by hori-ryota** [comment]
類語でどれを使えばいいのかが判断しやすくなるのと、略語も意味を取りやすくなるのと、日本語だけだと意味を掴みづらかった用語も判断しやすくなりやすいので非常にありがたいです！

**Comment by hori-ryota** [comment]
こちら、その後いかがでしょうか。近いうちに対応いただけるのであればライブラリの更新タイミングを合わせようと思っていまして、対応時期など見込みがあれば教えていただけますと幸いです。

**Comment by yasuyuki-nakazawa** [comment]
一旦、現状のリファレンスの列追加で英語版を準備したものの、内部で議論し、視認性が悪くなってしまうという点（スマホ版だと見切れが発生）、英語版ページを別途新設すべきという観点もあり、ペンドしております。

**Comment by hori-ryota** [comment]
ご回答ありがとうございます。リファレンスには非表示でOpenAPI仕様のみへの記載でも有用かと思います。ご検討願えると幸いです。

**Comment by hori-ryota** [comment]
ご検討いただいている英語名ですが、以下のような箇所での利用を目的としております。

[go\-kabusapi/structs\_gen\.go at master · hori\-ryota/go\-kabusapi](https://github.com/hori-ryota/go-kabusapi/blob/a3c1c55c2b07ed0a5f85d4ee6d54e27e5f0c7ccb/kabusapi/structs_gen.go#L2334)

[go\-kabusapi/structs\_gen\.go at master · hori\-ryota/go\-kabusapi](https://github.com/hori-ryota/go-kabusapi/blob/a3c1c55c2b07ed0a5f85d4ee6d54e27e5f0c7ccb/kabusapi/structs_gen.go#L1265)

これらはyamlの仕様書より自動生成しているため、英語名が定義されるとより適切な変数名を当てられるので嬉しいです。

なお、あまりにも長い英語名が生じた場合は生成時に略語などに変換することも検討できるので、まずは正しい英語名が付記されると助かります。

**Comment by yasuyuki-nakazawa** [comment]
お世話になります。
yamalより自動生成しているとのことですが、下記のような挿入だといかがですか。
コメントとして付加

取得する商品
          |定義値|説明|
          |-|-|
          |0|すべて|
          |1|現物|
          |2|信用|
          |3|先物|
          |4|OP|

   # 0 : すべて | ALL
   # 1 : 現物 | Cash
   # 2 : 信用 | Margin
   # 3 : 先物 | Future
   # 4 : OP | Option

**Comment by hori-ryota** [comment]
非常に助かります！

**Comment by yasuyuki-nakazawa** [comment]
少々お時間かかりますが、上記で対応します

**Comment by hori-ryota** [comment]
ありがとうございます！！
